OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -125,24 +128,59 @@ public void SetHardwareCursor(IHardwareCursor cursor)
 			}
 		}
 
-		class SDL2HardwareCursor : IHardwareCursor
+		sealed class SDL2HardwareCursor : IHardwareCursor
 		{
-			public readonly IntPtr Cursor;
-			readonly IntPtr surface;
+			public IntPtr Cursor { get; private set; }
+			IntPtr surface;
 
 			public SDL2HardwareCursor(Size size, byte[] data, int2 hotspot)
 			{
-				surface = SDL.SDL_CreateRGBSurface(0, size.Width, size.Height, 32, 0x00FF0000, 0x0000FF00, 0x000000FF, 0xFF000000);
+				try
+				{
+					surface = SDL.SDL_CreateRGBSurface(0, size.Width, size.Height, 32, 0x00FF0000, 0x0000FF00, 0x000000FF, 0xFF000000);
+					if (surface == IntPtr.Zero)
+						throw new InvalidDataException("Failed to create surface: {0}".F(SDL.SDL_GetError()));
+
+					var sur = (SDL2.SDL.SDL_Surface)Marshal.PtrToStructure(surface, typeof(SDL2.SDL.SDL_Surface));
+					Marshal.Copy(data, 0, sur.pixels, data.Length);
+
+					// This call very occasionally fails on Windows, but often works when retried.
+					for (var retries = 0; retries < 3 && Cursor == IntPtr.Zero; retries++)
+						Cursor = SDL.SDL_CreateColorCursor(surface, hotspot.X, hotspot.Y);
+					if (Cursor == IntPtr.Zero)
+						throw new InvalidDataException("Failed to create cursor: {0}".F(SDL.SDL_GetError()));
+				}
+				catch
+				{
+					Dispose();
+					throw;
+				}
+			}
 
-				var sur = (SDL2.SDL.SDL_Surface)Marshal.PtrToStructure(surface, typeof(SDL2.SDL.SDL_Surface));
-				Marshal.Copy(data, 0, sur.pixels, data.Length);
-				Cursor = SDL.SDL_CreateColorCursor(surface, hotspot.X, hotspot.Y);
+			~SDL2HardwareCursor()
+			{
+				Game.RunAfterTick(() => Dispose(false));
 			}
 
 			public void Dispose()
 			{
-				SDL.SDL_FreeCursor(Cursor);
-				SDL.SDL_FreeSurface(surface);
+				Game.RunAfterTick(() => Dispose(true));
+				GC.SuppressFinalize(this);
+			}
+
+			void Dispose(bool disposing)
```

Comment:
[Dispose pattern](https://msdn.microsoft.com/en-us/library/b1yfkh5e%28v=vs.110%29.aspx) for reference.

It's so you can do 
```
if (disposing)
    foo.Dispose();
```

It just happens there isn't anything that needs it in this case. Whilst that's a little annoying I generally find it preferable to not mess with the recommended dispose pattern too much because it's complicated enough as it is without introducing your own variations.